### PR TITLE
Fix ScanIterator unable to read from adjacent empty regions

### PR DESCRIPTION
**Pull request by birdstorm**
A missing PR.

### CODE REVIEW

**Comment by birdstorm** [comment]
tests are failing because of https://github.com/pingcap/tidb/issues/8258 You may ignore them.

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java [hunk 1]
```diff
@@ -67,10 +67,8 @@ private boolean loadCache() {
       BackOffer backOffer = ConcreteBackOffer.newScannerNextMaxBackOff();
       currentCache = client.scan(backOffer, startKey, version);
       // currentCache is null means no keys found, whereas currentCache is empty means no values
-      // found
-      // the difference lies in whether to continue scanning, because chances are that the same key
-      // is
-      // split in another region because of pending entries, region split, e.t.c.
+      // found the difference lies in whether to continue scanning, because chances are that the
+      // same key is split in another region because of pending entries, region split, e.t.c.
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java]: s/split/spliting

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java]: split is its own passive voice...

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java [hunk 2]
```diff
@@ -110,21 +108,21 @@ public boolean hasNext() {
...
 
   @Override
   public Kvrpcpb.KvPair next() {
+    if (currentCache == null) {
+      throw new TiClientInternalException("cache is null, read data failed");
+    }
     Kvrpcpb.KvPair kv = getCurrent();
-    if (kv == null) {
-      // cache drained
+    while (currentCache != null && kv == null) {
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java]: replace with do while？

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java [hunk 3]
```diff
@@ -110,21 +108,21 @@ public boolean hasNext() {
...
-      return currentCache.get(index++);
-    }
-    return null;
+    return currentCache.get(index++);
   }
 
   @Override
   public Kvrpcpb.KvPair next() {
+    if (currentCache == null) {
+      throw new TiClientInternalException("cache is null, read data failed");
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java]: It seems not necessary. We already checked `currentCache`'s value at `isCacheDrained`

File: tikv-client/src/test/java/com/pingcap/tikv/txn/LockResolverTest.java [hunk 4]
```diff
@@ -205,8 +205,7 @@ private boolean commit(long startTS, long commitTS, List<ByteString> keys) {
               resp -> resp.hasRegionError() ? resp.getRegionError() : null);
 
       CommitResponse resp =
-          client.callWithRetry(
-              backOffer, TikvGrpc.METHOD_KV_COMMIT, factory, handler);
+          client.callWithRetry(backOffer, TikvGrpc.METHOD_KV_COMMIT, factory, handler);
```
Comment (zhexuany) [inline tikv-client/src/test/java/com/pingcap/tikv/txn/LockResolverTest.java]: format issue？

Comment (birdstorm) [reply tikv-client/src/test/java/com/pingcap/tikv/txn/LockResolverTest.java]: yes

**Pull request merged**